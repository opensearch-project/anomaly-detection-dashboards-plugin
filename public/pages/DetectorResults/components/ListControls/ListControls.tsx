/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { EuiFlexGroup, EuiFlexItem, EuiPagination } from '@elastic/eui';

interface ListControlsProps {
  activePage: number;
  pageCount: number;
  onPageClick: (pageNumber: number) => void;
}
export const ListControls = (props: ListControlsProps) => (
  <EuiFlexGroup
    style={{ padding: '0px 5px' }}
    alignItems="center"
    justifyContent="flexEnd"
  >
    {props.pageCount > 1 ? (
      <EuiFlexItem grow={false} style={{ justifyContent: 'center' }}>
        <EuiPagination
          pageCount={props.pageCount}
          activePage={props.activePage}
          onPageClick={props.onPageClick}
          data-test-subj="anomaliesPageControls"
        />
      </EuiFlexItem>
    ) : null}
  </EuiFlexGroup>
);
