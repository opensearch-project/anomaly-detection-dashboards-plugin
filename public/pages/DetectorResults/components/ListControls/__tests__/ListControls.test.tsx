/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ListControls } from '../ListControls';

describe('<ListControls /> spec', () => {
  const defaultProps = {
    activePage: 1,
    pageCount: 10,
    search: '',
    onSearchChange: jest.fn(),
    onPageClick: jest.fn(),
  };
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('Empty results', () => {
    test('renders component with empty message', async () => {
      const { container } = render(<ListControls {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    test('pagination should be visible if pages count more than 1', async () => {
      const { queryByTestId } = render(
        <ListControls {...defaultProps} pageCount={2} />
      );
      expect(queryByTestId('anomaliesPageControls')).not.toBeNull();
    });
    test('should call onPageClick when click on the page', async () => {
      const { getByTestId } = render(<ListControls {...defaultProps} />);
      fireEvent.click(getByTestId('pagination-button-3'));
      expect(defaultProps.onPageClick).toHaveBeenCalledTimes(1);
    });
  });
});
