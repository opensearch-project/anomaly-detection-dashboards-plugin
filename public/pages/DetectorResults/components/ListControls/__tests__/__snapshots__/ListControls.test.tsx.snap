// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListControls /> spec Empty results renders component with empty message 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  style="padding: 0px 5px;"
>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
    style="justify-content: center;"
  >
    <nav
      class="euiPagination"
      data-test-subj="anomaliesPageControls"
    >
      <button
        aria-label="Previous page, 1"
        class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall"
        data-test-subj="pagination-button-previous"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
          />
        </svg>
      </button>
      <ul
        class="euiPagination__list"
      >
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 1 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-0"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                1
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-current="true"
            aria-label="Page 2 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiButtonEmpty-isDisabled euiPaginationButton euiPaginationButton-isActive euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-1"
            disabled=""
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                2
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 3 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-2"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                3
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 4 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-3"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                4
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 5 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-4"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                5
              </span>
            </span>
          </button>
        </li>
        <li
          aria-label="Skipping pages 6 to 9"
          class="euiPaginationButton-isPlaceholder euiPagination__item"
        >
          â€¦
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 10 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-9"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                10
              </span>
            </span>
          </button>
        </li>
      </ul>
      <button
        aria-label="Next page, 3"
        class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall"
        data-test-subj="pagination-button-next"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
          />
        </svg>
      </button>
    </nav>
  </div>
</div>
`;
